### PR TITLE
♻️ [RUM-1039] Harmonize view tests

### PR DESCRIPTION
**Pull request by amortemousque**
## Motivation

Following the [view metrics splitting](https://github.com/DataDog/browser-sdk/pull/2386), this PR reorganize the view tracking tests so that every spec files tests its related module.

## Changes

- Cleanly stop INP tests
- Move CLS support check in trackCommonViewMetrics module
- Scope trackCumulativeLayoutShift tests
- Scope trackLoadingTime tests
- Scope trackViewEventCounts tests
- Reorganize trackViews tests

## Testing

- [ ] Local
- [ ] Staging
- [x] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

**Review by BenoitZugmeyer — APPROVED** [review]

File: packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts [hunk 1]
```diff
@@ -36,114 +47,85 @@ describe('trackLoadingTime', () => {
...
   })
 
   it('should use loadEventEnd for initial view when having no activity', () => {
+    loadType = ViewLoadingType.INITIAL_LOAD
     const { lifeCycle, clock } = setupBuilder.build()
-    const { getViewUpdate, getViewUpdateCount } = viewTest
 
-    expect(getViewUpdateCount()).toEqual(1)
     const entry = createPerformanceEntry(RumPerformanceEntryType.NAVIGATION)
     lifeCycle.notify(LifeCycleEventType.PERFORMANCE_ENTRIES_COLLECTED, [entry])
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts:71]: **💬 suggestion:** ‏remove this useless `lifeCycle.notify`

File: packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts [hunk 2]
```diff
@@ -36,114 +47,85 @@ describe('trackLoadingTime', () => {
...
     // to make the test fail if the clock gap is not correctly taken into account
     const CLOCK_GAP = (LOAD_EVENT_AFTER_ACTIVITY_TIMING - BEFORE_PAGE_ACTIVITY_VALIDATION_DELAY + 1) as Duration
 
-    setupBuilder.clock!.tick(CLOCK_GAP)
-
-    const { lifeCycle, domMutationObservable, clock } = setupBuilder.build()
-    const { getViewUpdate, getViewUpdateCount } = viewTest
-
-    expect(getViewUpdateCount()).toEqual(1)
+    clock.tick(CLOCK_GAP)
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts:118]: **💬 suggestion:** ‏According to the comment above, the `clock.tick(CLOCK_GAP)` should be done before `setupBuilder.build()`. Maybe revert to the previous implem?

Comment (amortemousque) [reply packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts:118]: Hum, I checked and even with the previous version of the test if you move `clock.tick(CLOCK_GAP)` after `setupBuilder.build()` it still works..

Comment (BenoitZugmeyer) [reply packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts:118]: Yes. The point of the test is to make sure the loading time is relative to `timeOrigin` and not relative to the time `trackLoadingTime` was started. The `clock.tick(CLOCK_GAP)` was there to emulate a `trackLoadingTime()` invoked a bit after `timeOrigin`.

Comment (amortemousque) [reply packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts]: I fixed the test

File: packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts [hunk 3]
```diff
@@ -36,114 +46,80 @@ describe('trackLoadingTime', () => {
...
-      createPerformanceEntry(RumPerformanceEntryType.NAVIGATION, {
-        loadEventEnd: LOAD_EVENT_BEFORE_ACTIVITY_TIMING,
-      }),
-    ])
+    setLoadEvent(LOAD_EVENT_BEFORE_ACTIVITY_TIMING)
 
     domMutationObservable.notify()
     clock.tick(AFTER_PAGE_ACTIVITY_END_DELAY)
-
     clock.tick(THROTTLE_VIEW_UPDATE_PERIOD)
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/view/viewMetrics/trackLoadingTime.spec.ts:121]: **🥜 nitpick:** ‏this `THROTTLE_VIEW_UPDATE_PERIOD` could probably be removed

```suggestion
```

**Review by BenoitZugmeyer — APPROVED** [review]

**Review by BenoitZugmeyer — APPROVED** [review]

**Pull request merged**